api/v1alpha1/amaltheasession_types.go

```diff
@@ -269,6 +275,184 @@ type AmaltheaSession struct {
 	Status AmaltheaSessionStatus `json:"status,omitempty"`
 }
 
+type AmaltheaChildren struct {
+	Ingress     *networkingv1.Ingress
+	Service     v1.Service
+	StatefulSet appsv1.StatefulSet
+	PVC         v1.PersistentVolumeClaim
+}
+
+func (cr *AmaltheaSession) OwnerReference() metav1.OwnerReference {
+	gvk := cr.GroupVersionKind()
+	return metav1.OwnerReference{
+		APIVersion:         gvk.GroupVersion().String(),
+		Kind:               gvk.Kind,
+		Name:               cr.ObjectMeta.Name,
+		BlockOwnerDeletion: pointer.Bool(true),
+		Controller:         pointer.Bool(true),
+		UID:                cr.GetObjectMeta().GetUID(),
+	}
+}
+
+func (cr *AmaltheaSession) Children() AmaltheaChildren {
+	return AmaltheaChildren{
+		StatefulSet: statefulSetForAmaltheaSession(cr),
+		Service:     serviceForAmaltheaSession(cr),
+		Ingress:     ingressForAmaltheaSession(cr),
+	}
+}
+
+// statefulSetForAmaltheaSession returns a AmaltheaSession StatefulSet object
+func statefulSetForAmaltheaSession(
```

Comment:
@olevski Should we also make these free functions methods of `AmaltheaSession` struct ?

One reason would be to make their call more intuitive when outside of this file when not needing all of the children for example when checking the pod created by the `StatefulSet`

Reply:
Yes that would be really nice! After that I will approve and we can merge.

Reply:
Done !